File: problems/basic/461/solution.cpp
// Accepted: 0ms (100.00%), 6.28% (76.44%)

class Solution {
public:
    int hammingDistance(int x, int y) {
        int xored = x ^ y;
        int dist = 0;
        while (xored > 0) {
            if ((xored & 1) == 1) {
                dist++;
            }
            xored >>= 1;
        }
        return dist;
    }
};

File: problems/basic/461/solution.py
# Accepted: 34ms (81.50%), 17.18MB (34.65%)

class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        xored = x ^ y
        dist = 0
        while xored > 0:
            if (xored & 1) == 1:
                dist += 1
            xored >>= 1
        return dist
